client/ayon_maya/plugins/publish/validate_skeleton_top_group_hierarchy.py

```diff
@@ -40,3 +40,21 @@ def get_top_hierarchy(self, targets):
             target for target in targets if target.count("|") > 2
         ]
         return non_top_hierarchy_list
+
+class ValidateAnimatedRigTopGroupHierarchy(ValidateSkeletonTopGroupHierarchy):
+    order = ValidateContentsOrder + 0.05
+    label = "Animated Rig Top Group Hierarchy"
+    families = ["animation.fbx"]
+    optional = True
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+        invalid = []
+        skeleton_mesh_data = instance.data("animated_skeleton", [])
+        if skeleton_mesh_data:
+            invalid = self.get_top_hierarchy(skeleton_mesh_data)
+            if invalid:
+                raise PublishValidationError(
+                    "The skeletonAnim_SET includes the object which "
+                    "is not at the top hierarchy: {}".format(invalid))
```

Comment:
So it is valid if there is no animated skeleton? Which was the case that I faced where it errored on integrate?

Also, the inheritance from one another here is quite redundant - it works just as well if we separate `get_top_hierarchy` into a separate function.

Also, I'm quite confident that you get the exact same results from `cmds.ls(targets, assemblies=True, long=True)` ;) which should return only assemblies (top level root nodes)